src/Board.hs

```diff
@@ -86,10 +88,10 @@ slideNumber number board =
           validateAndSlideNumber (Just blankPosition) (Just numberPosition)
                 | nextTo blankPosition numberPosition = Right $ swapNumberAndBlank number board
                 | otherwise                           = Left NumberNotAdjacentToBlank
-          nextTo (Position bx by) (Position nx ny)    =
-                (nx == bx && (ny == by-1 || ny == by+1)) ||
-                (ny == by && (nx == bx-1 || nx == bx+1))
-          swapNumberAndBlank number (Board d layout)  = Board d $ map swappy layout
+
+          nextTo (Position bx by) (Position nx ny)    = abs (bx - nx) + abs (by - ny) == 1
```

Comment:
My code is a bit shorter, but your code is more straightforward. So this change is not necessarily an improvement.